app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,210 @@
+
+    def test_update_recipe_new_ingredients_successful(self):
+        """Test updating recipe with list of ingredients"""
+        recipe = sample_recipe(name='Cheeseburger', description='Yummy')
+        payload = {
+            'id': recipe.id,
+            'name': recipe.name,
+            'description': recipe.description,
+            'ingredients': [
+                {'name': 'Cheese'}, {'name': 'Bun'}, {'name': 'Meat'}
+            ]
+        }
+        url = detail_url(recipe.id)
+        res = self.client.put(url, payload, format='json')
+        recipe.refresh_from_db()
+        serializer = RecipeSerializer(recipe)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(serializer.data, res.data)
+
+    def test_update_recipe_all_fields_successful(self):
+        """Test updating recipe with replacement ingredients"""
+
+        """Create initial recipe with ingredients first"""
+        recipe = sample_recipe(name='Cheeseburger', description='Yummy')
+        payload = {
+            'id': recipe.id,
+            'name': recipe.name,
+            'description': recipe.description,
+            'ingredients': [
+                {'name': 'Cheese'}, {'name': 'Bun'}, {'name': 'Meat'}
+            ]
+        }
+        url = detail_url(recipe.id)
+        self.client.put(url, payload, format='json')
+
+        """Update recipe with new data"""
+        payload2 = {
+            'id': recipe.id,
+            'name': 'Ice Cream',
+            'description': 'Cold',
+            'ingredients': [{'name': 'Vanilla'}]
+        }
+
+        res2 = self.client.put(url, payload2, format='json')
+        recipe.refresh_from_db()
+
+        self.assertEqual(res2.status_code, status.HTTP_200_OK)
+        self.assertEqual(recipe.name, payload2['name'])
+        self.assertEqual(recipe.description, payload2['description'])
+        self.assertEqual(
+            len(recipe.ingredients.all()),
+            len(payload2['ingredients'])
+        )
+
+        serializer = RecipeSerializer(recipe)
+        self.assertEqual(serializer.data, res2.data)
+
+    def test_update_invalid_recipe(self):
+        """Test updating invalid recipe"""
+        url = detail_url(-1)
+        res = self.client.put(url, {}, format='json')
+        self.assertEqual(res.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete_recipe_successful(self):
+        """Test deleting recipe also delete ingredients"""
+
+        """Create initial recipe with ingredients first"""
+        recipe = sample_recipe(name='Cheeseburger', description='Yummy')
+        payload = {
+            'id': recipe.id,
+            'name': recipe.name,
+            'description': recipe.description,
+            'ingredients': [
+                {'name': 'Cheese'}, {'name': 'Bun'}, {'name': 'Meat'}
+            ]
+        }
+        url = detail_url(recipe.id)
+        self.client.put(url, payload, format='json')
```

Comment:
why do you do this put in the middle to test the deletion?

Reply:
Agreed, this put is unnecessary for the deletion test.